projects/cega/evm.js

```diff
@@ -0,0 +1,116 @@
+        const FCNProductAddress = await sdk.api.abi.call({
+            target: CEGA_STATE,
+            abi: abi.products,
+            params: product
+        })
+    FCNProducts.push(FCNProductAddress.output);
+    }
+
+    let LOVProducts = [];
+    for(const product of LOVProductNames){
+        const LOVProductAddress = await sdk.api.abi.call({
+            target: CEGA_STATE,
+            abi: abi.products,
+            params: product
+        })
+        LOVProducts.push(LOVProductAddress.output);
+    }
+    return [FCNProducts, LOVProducts];
+}
+
+
+async function getSumFCNProductDeposits(fcnProducts){
+    let totalBalance = 0; 
+    for (const product of fcnProducts){
+        const result = await sdk.api.abi.call({
+            target: product,
+            abi: abi.sumVaultUnderlyingAmounts,
+        })
+        const amount = parseInt(result.output)
+        totalBalance += amount; 
+    }
+    return (totalBalance / usdcDecimals);
+}
+
+async function getSumFCNProductQueuedDeposits(fcnProducts){
+    let totalBalance = 0; 
+    for (const product of fcnProducts){
+        const result = await sdk.api.abi.call({
+            target: product,
+            abi: abi.queuedDepositsTotalAmount,
+        })
+        const amount = parseInt(result.output)
+        totalBalance += amount; 
+    }
+    return (totalBalance / usdcDecimals);
+}
+
+async function getSumLOVProductDeposits(lovProducts){
+    let totalBalance = 0; 
+    for (const product of lovProducts){
+        for (let i = 2; i < maxLeverage; i++){
+        const result = await sdk.api.abi.call({
+            target: CEGA_PRODUCT_VIEWER,
+            abi: abi.getLOVVaultMetadata, 
+            params: [product , i],
+        })
+        const amount = result.output.reduce((total, arg) => total + parseInt(arg.underlyingAmount), 0);
+        totalBalance += amount; 
+        }
+    }
+    return totalBalance / usdcDecimals;
+}
+
+async function getSumLOVProductQueuedDeposits(lovProducts){
+    let totalBalance = 0; 
+    for (const product of lovProducts){
+        for (let i = 2; i < maxLeverage; i++){
+        const result = await sdk.api.abi.call({
+            target: CEGA_PRODUCT_VIEWER,
+            abi: abi.getLOVProductQueuedDeposits, 
+            params: [product , i],
+        })
+        totalBalance += parseInt(result.output);
+        }
+    }
+    return totalBalance / usdcDecimals;
+}
+
+async function getEthereumTvl(){
+    const [fcnProducts, lovProducts] = await getProducts();
+    const results = await Promise.all([
+        getSumFCNProductDeposits(fcnProducts),
+        getSumFCNProductQueuedDeposits(fcnProducts),
+        getSumLOVProductDeposits(lovProducts),
+        getSumLOVProductQueuedDeposits(lovProducts)
+    ]);
+    const sum = results.reduce((total, currentValue) => total + currentValue, 0);
+    return sum 
+}
+
+
+module.exports = {
+    ethereum: {
+       tvl: getEthereumTvl().then((result) => { console.log(result)})
```

Comment:
this is incorrect, it should be an async function that returns a balances object

Reply:
same for solana's side